File: src/main/java/edu/kruchon/natural/lang/parser/codegeneration/KotlinClass.java
package edu.kruchon.natural.lang.parser.codegeneration;

import java.util.HashSet;
import java.util.Set;

public class KotlinClass {
    private final String name;
    // todo take in account the case with duplicated field names
    private final Set<KotlinClass> fields;

    public KotlinClass(String name) {
        this.name = name;
        this.fields = new HashSet<>();
    }

    public KotlinClass(String name, Set<KotlinClass> fields) {
        this.name = name;
        this.fields = new HashSet<>(fields);
    }

    public String getName() {
        return name;
    }

    public Set<KotlinClass> getFields() {
        return fields;
    }
}

File: src/main/kotlin/edu/kruchon/natural/lang/parser/codegeneration/KotlinClass.kt
package edu.kruchon.natural.lang.parser.codegeneration

class KotlinClass {
    val name: String

    // todo take in account the case with duplicated field names
    val fields: Set<KotlinClass>

    constructor(name: String) {
        this.name = name
        fields = setOf()
    }

    constructor(name: String, fields: Set<KotlinClass>) {
        this.name = name
        this.fields = fields
    }
}